### ISSUE TITLE
Allow late stand submission discussion

### ISSUE THREAD

**Issue by rahul2240** [openSUSE/osem#2534]
**I'm submitting a ..**
- [ ] Bug Report
- [ ] Feature Request

My approach is - 
- A form to enter emails to invite those people.
- if user exists then send a mail otherwise send an invitation mail.
- I will add an invited_ids column to conference that will store all the user id of people that are allowed to submit late stand for that particular conference.
- Add in the ability to allow users with user id same as present in invited_ids column.

**Comment by differentreality** [comment]
I am confused, is this about late submissions or inviting users to become booth responsibles?!

**Comment by rahul2240** [comment]
@differentreality  Both. After the booth request period is over then organizers can invite people to register, once they do that then request booth option will be available to these invited people.

**Comment by differentreality** [comment]
So invited people can have the extra privilege to request a booth after the deadline? 
How do we know if they were invited at any given time to be a booth responsible or if they were invited to be granted the extra privilege? 
A booth responsible can be added/invited at any time.

**Comment by rahul2240** [comment]
@differentreality , Yes they have the extra privilege to request a booth after the deadline. 
I will add one more attribute to the conference, it will be for entering the number of days by the organizer, invitation expires after that number of days.
Yes booth responsible can be added/invited at any time.

**Comment by rahul2240** [comment]
@differentreality @Ana06 I could not find a way to invite people for a particular conference using devise invitable gem. I have also asked the creators, and they said there is no way yet.
So, I am thinking of setting invitation token expire time to constant say 1 or 2 weeks and create a new invitations model that will hold a user id, conference id and invite expire field.
Whenever a user signs up or logs in, this invitation model will be checked to find "whether user is allowed to submit a booth request or not'.
This approach has advantages like we can invite a user multiple times and for multiple conferences, also we can update the model in future according to the needs.

**Comment by differentreality** [comment]
@rahul2240 to make sure we all understand the same things, please write down the steps we expect users to follow and show us the relationships and sample db tables of what you are suggesting.

**Comment by rahul2240** [comment]
@differentreality , here is what I am thinking.

What I want to do:
1. Admin will visit a invite_booth_path.
2. There will be a form with email id field(multiple emails) and number of days to invite those email users. 
3. Admin will enter email ids and number of days to invite those emails and submit the form.
4. If the invited user registers within that "number of days" after the day user is invited then user can submit booth request (request booth button visible). Otherwise, request booth button will not be visible.

My approach for implementation:
1. Create a form with email and "number of days" field.
2. A new controller i.e "invites controller" for the form data.
3. A new model i.e "invites model" that will store the following fields.
    a. user id of the invited user
    b. number of days to invite
    c. conference_id of the conference for which the user is invited.

**Comment by kergon** [comment]
On Mon, Jun 17, 2019 at 11:04:52AM -0700, Rahul wrote:
>        b. number of days to invite

I suggest a focus on the new deadline might work better than having that
as a derived quantity to be calculated: everyone involved is thinking in
terms of the new date that applies, rather than the number of days away
that is relative to some other date.

Alasdair

**Comment by rahul2240** [comment]
@kergon Thanks for the suggestion, you are right. Date will be better.

**Comment by differentreality** [comment]
@rahul2240 so we would need two separate new controllers and db tables for invitations regarding booths and tracks?

**Comment by rahul2240** [comment]
We will need a db table for invitations regarding booth and tracks. We can use a separate controller to keep the invitation code separate but we can use concerns also.
